### PR TITLE
Lecture2

### PR DESCRIPTION
**Pull request by RomanKukui**

### CODE REVIEW

File: Lecture_2/Exercise_3/list.c [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+			return list_ptr;
+		} else {
+			list_ptr = list_ptr->next;
+			i++;
+		}
+	}
+	return NULL;		// unexpected end of list
+}
+
+void* ListIs(list *list_ptr, uint32_t index)
```
Comment (PavloKovalchuk1) [inline Lecture_2/Exercise_3/list.c:78]: The idea of Listls function was to print all values from list (just like an ls command in linux terminal), not to read a single value.

But it's ok as it was not very clear, and you've implemented the "print all" functionality within the menu of main.c.

File: Lecture_2/Exercise_3/list.c [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	if (list_ptr == NULL) {
+		curr_item->prev = NULL;
+		curr_item->next = NULL;
+
+		*root_ptr = curr_item;	// update root by first
+	} else {
+		list_ptr = ListGetLast(list_ptr);
+
+		curr_item->prev = list_ptr;
+		curr_item->next = list_ptr->next;
```
Comment (PavloKovalchuk1) [inline Lecture_2/Exercise_3/list.c]: Is there a way that on this line list_ptr->next will not be NULL? If no, then it makes sense to just write
`curr_item->next = NULL;`
for clarity.

**Review by PavloKovalchuk1 — APPROVED** [review]
Looks very good overall. You can merge it to master.

File: Lecture_2/Exercise_3/list.c (hunk 2 above)
Comment (RomanKukui) [reply Lecture_2/Exercise_3/list.c]: Pavlo, you are right) curr_item->next = NULL; is better/more clear in this case. Thanks!

**Pull request merged**